Frameworks/VMFleet/VMFleet.psm1

```diff
@@ -8665,4 +9136,152 @@ function SetCacheBehavior
             Remove-PSSession -Session $s
         }
     }
+}
+
+function Get-Salt {
+    $salt = ([char]'a'..[char]'z' + [char]'A'..[char]'Z' + [char]'0'..[char]'9' | Get-Random -Count 4 |% { [char]$_ }) -join ''
+    Write-Host "Generated Salt: $salt"
+    return $salt;
+}
+
+function Get-ArcConfig {
+    [CmdletBinding()]
+    param(
+        [Parameter()]
+        [string]
+        $Cluster = "."
+    )
+
+    $controlPath = Get-FleetPath -PathType Control $Cluster
+    $f = Join-Path $controlPath "arc.json"
+    if (-not (Test-Path $f)) {
+        Write-Host "arc.json does not exist. Current Arc mode: Disabled."
+        return $null
+    }
+
+    $arcConfig = Get-Content -Path $f -Raw | ConvertFrom-Json
+    return $arcConfig
+}
+
+function Set-ArcConfig {
+    [CmdletBinding()]
+    param(
+        [Parameter()]
+        [string]$ResourceGroup,
+
+        [Parameter()]       
+        [string]$AzureRegistrationUser,
+
+        [Parameter()]
+        [string]$AzureRegistrationPassword,
+        
+        [Parameter()]
+        [string]$StoragePathCsv,
+
+        [Parameter()]
+        [string]$StoragePathName = "arcVmFleetStoragePath",
+
+        [Parameter()]
+        [string]$ImageName = "arcVmFleetImage",
+        
+        [Parameter()]
+        [bool]
+        $Enabled = $false,
+
+        [Parameter()]
+        [switch]
+        $ResetSalt,
+
+        [Parameter()]
+        [string]
+        $Cluster = "."
+    )
+
+    if($PSBoundParameters.Count -eq 0){
+        Write-Host "Set atleast one property."
```

Comment:
"at least" and this seems like an error condition - Write-Error?

Reply:
Note that Test-ArcConfig will do that on error in the normal flow, below.

Reply:
this is added as a fail fast mechanism as it does not make sense to move forward beyond this point if no parameter is provided by user. Test-ArcConfig helps validate if all mandatory parameters are available to vmfleet by checking existing arc.json if any and current user provided parameters.